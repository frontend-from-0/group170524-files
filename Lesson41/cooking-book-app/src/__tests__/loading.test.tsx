import { render, screen } from "@testing-library/react";
import Loading from "@/app/loading";

describe("Loading page", () => {
  it("renders a heading", () => {
    render(<Loading />);

    const heading = screen.getByRole("heading", { level: 1 });

    expect(heading).toBeInTheDocument();
  });

  it("renders placeholder items", () => {
    render(<Loading />);

    const placeholderList = screen.getByTestId("placeholder-list");
    const placeholderItems = screen.getAllByTestId("placeholder-item");

    expect(placeholderList).toBeInTheDocument();
    expect(placeholderItems.length).toBe(16);
  });

  describe("Snapshot", () => {
    it("renders loading page unchanged", () => {
      const { container } = render(<Loading />);
      expect(container).toMatchSnapshot();
    });
  })
});

// test('Loading page is rendered correctly', () => {
//   render(<Loading />);

//   const heading = screen.getByRole('heading', { level: 1 });

//   expect(heading).toBeInTheDocument();
// })
