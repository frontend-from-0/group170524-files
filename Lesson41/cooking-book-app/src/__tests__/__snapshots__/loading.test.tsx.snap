// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading page Snapshot renders loading page unchanged 1`] = `
<div>
  <div
    class="text-center my-8"
  >
    <h1>
      Home Page
    </h1>
    <ul
      class="max-w-6xl mx-auto flex flex-wrap gap-6 justify-between"
      data-testid="placeholder-list"
    >
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
      <li
        class="max-w-3xs my-4"
        data-testid="placeholder-item"
      >
        <div
          class="w-48 h-6 bg-gray-100 my-2 mx-auto rounded"
        />
        <div
          class="w-3xs h-64 bg-gray-100"
        />
      </li>
    </ul>
  </div>
</div>
`;
